storage/filesystem/index.go

```diff
@@ -20,8 +20,14 @@ func (s *IndexStorage) SetIndex(idx *index.Index) (err error) {
 	}
 
 	defer ioutil.CheckClose(f, &err)
+	bw := bufio.NewWriter(f)
```

Comment:
Have you found substantial difference using a buffered writer to save the index?

Reply:
local benchmark result. reverted buffered writer.
```
name          old time/op    new time/op    delta
PlainClone-4      173s ± 5%      251s ± 9%  +44.84%  (p=0.008 n=5+5)

name          old alloc/op   new alloc/op   delta
PlainClone-4    29.8GB ± 0%    29.4GB ± 0%   -1.12%  (p=0.008 n=5+5)

name          old allocs/op  new allocs/op  delta
PlainClone-4      114M ± 0%      114M ± 0%   -0.04%  (p=0.008 n=5+5)
```